processor/spanmetricsprocessor/cache/cache_test.go

```
@@ -0,0 +1,153 @@
+			args: args{
+				size: -1,
+			},
+			wantErr: true,
+		},
+		{
+			name: "create a new Cache with length 0",
+			args: args{
+				size: 0,
+			},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			_, err := NewCache(tt.args.size)
+			if tt.wantErr {
+				assert.Error(t, err)
+				return
+			}
+			assert.NoError(t, err)
+		})
+	}
+}
+
+func TestCache_Get(t *testing.T) {
+	tests := []struct {
+		name         string
+		lruCache     func() *Cache
+		evictedItems map[interface{}]interface{}
+		key          interface{}
+		wantValue    interface{}
+		wantOk       bool
+	}{
+		{
+			name: "if key is not found in LRUCache, will get key from evictedItems",
+			lruCache: func() *Cache {
+				cache, _ := NewCache(1)
+				cache.evictedItems["key"] = "val"
+				return cache
+			},
+			key:       "key",
+			wantValue: "val",
+			wantOk:    true,
+		},
+		{
+			name: "if key is found in LRUCache, return the found item",
+			lruCache: func() *Cache {
+				cache, _ := NewCache(1)
+				cache.Add("key", "val_from_LRU")
+				cache.evictedItems["key"] = "val_from_evicted_items"
+				return cache
+			},
+			key:       "key",
+			wantValue: "val_from_LRU",
+			wantOk:    true,
+		},
+		{
+			name: "if key is not found either in LRUCache or evicted items, return nothing",
+			lruCache: func() *Cache {
+				cache, _ := NewCache(1)
+				return cache
+			},
+			key:       "key",
+			wantValue: nil,
+			wantOk:    false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			c := tt.lruCache()
+			gotValue, gotOk := c.Get(tt.key)
+			if !reflect.DeepEqual(gotValue, tt.wantValue) {
+				t.Errorf("Get() gotValue = %v, want %v", gotValue, tt.wantValue)
+			}
+			if gotOk != tt.wantOk {
+				t.Errorf("Get() gotOk = %v, want %v", gotOk, tt.wantOk)
+			}
+		})
+	}
+}
+
+func TestCache_RemoveEvictedItems(t *testing.T) {
+	tests := []struct {
+		name     string
+		lruCache func() *Cache
```

```suggestion
lruCache func(t *testing.T) *cache {
    cache, err := NewCache(1)
    require.NoError(t, err, "Must not error when creating the cache")
   // .. The remaining set up.
}
```

It looks like a code smell ignoring the error within the test setup code. While I understand that currently it is not possible to error, we should ensure that it is checked incase another developer makes changes that break the current contract.